tools/cli/domainCommands.go

```diff
@@ -76,6 +76,7 @@ func newDomainCLI(
 			return c.String(FlagDestinationAddress)
 		}).ServerFrontendClient(c)
 	} else {
+		d.frontendClient = initializeFrontendClient(c)
```

Comment:
you need to initialize destinationClient as well. And you can refactor the code here a little.

```
	d := &domainCLIImpl{}
	d.frontendClient = initializeFrontendClient(c)
	d.destinationClient = newClientFactory(func(c *cli.Context) string {
		return c.String(FlagDestinationAddress)
	}).ServerFrontendClient(c)
	if isAdminMode {
		d.frontendAdminClient = initializeFrontendAdminClient(c)
		d.destinationAdminClient = newClientFactory(func(c *cli.Context) string {
			return c.String(FlagDestinationAddress)
		}).ServerAdminClient(c)
		d.domainHandler = initializeAdminDomainHandler(c)
	}
	return d
```